src/snowflake/cli/plugins/nativeapp/commands.py

```diff
@@ -285,3 +285,17 @@ def app_deploy(
     return MessageResult(
         f"Deployed successfully. Application package and stage are up-to-date."
     )
+
+
+@app.command("validate", requires_connection=True)
+@with_project_definition("native_app")
+def app_validate(**options):
+    """
+    Validates a deployed Native App's setup script.
+    """
+    manager = NativeAppManager(
+        project_definition=cli_context.project_definition,
+        project_root=cli_context.project_root,
+    )
+    manager.validate()
```

Comment:
There are three commands that can create an application package today: `snow app deploy`, `snow app run`, `snow app version create`. These involve creating package, uploading files to stage, and then some additional steps. Should we add `manager.validate()` to these three commands as well once they finish the second step of upload? We can save user's time by erroring out early before they have to wait further for Snowflake operations. 

Can discuss as a team.

Reply:
Yeah that can definitely be done, this method should be callable anywhere. Should I do this in this PR or in a followup?

Reply:
If it is as straightforward as adding `manager.validate()` to the three methods, then we should do it here and mention the new validation step in the three commands in the release notes.